src/category_theory/limits/has_limits.lean

```diff
@@ -90,22 +91,32 @@ variables (J C)
 class has_limits_of_shape : Prop :=
 (has_limit : Π F : J ⥤ C, has_limit F . tactic.apply_instance)
 
-/-- `C` has all (small) limits if it has limits of every shape. -/
-class has_limits : Prop :=
+/--
+`C` has all limits of size `v u` (`has_limits_of_size.{v u} C`)
+if it has limits of every shape `J : Type u` with `[category.{v} J]`.
+-/
+class has_limits_of_size : Prop :=
 (has_limits_of_shape :
-  Π (J : Type v) [𝒥 : small_category J], has_limits_of_shape J C . tactic.apply_instance)
+  Π (J : Type u₁) [𝒥 : category.{v₁} J], has_limits_of_shape J C . tactic.apply_instance)
+
+/-- `C` has all (small) limits if it has limits of every shape that is as big as its hom-sets. -/
+abbreviation has_limits (C : Type u) [category.{v} C] : Prop := has_limits_of_size.{v v} C
+
+lemma has_limits.has_limits_of_shape {C : Type u} [category.{v} C] [has_limits C]
+  (J : Type v) [category.{v} J] :
+  has_limits_of_shape J C := has_limits_of_size.has_limits_of_shape J
```

Comment:
This (and its dual) is the only definitional change in this PR.
`has_limits` still means about the same thing, but lean seems to lose some ability to infer some types here and there and thus type and universe annotations needed to be added here and there.